modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/ranges.scala

```diff
@@ -0,0 +1,45 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.syntax.all._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.node.shared.domain.seedlist.Range.ValidationResult
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+
+case class InvalidInput(str: String) extends NoStackTrace {
+  override def getMessage = s"Invalid input: $str"
+}
+
+object ranges {
+
+  private val delimiter = ";"
+
+  def decode(input: String): ValidationResult[List[Range]] =
+    input
+      .split(delimiter)
+      .toList
+      .traverse(Range.decode)
+      .map(mergeOverlaps)
+
+  private def mergeOverlaps(ranges: Seq[Range]): List[Range] =
```

Comment:
you can use `List[Range]` as input, since you already construct a list of ranges in the previous step